mne/channels/montage.py

```diff
@@ -624,6 +633,45 @@ def read_dig_montage(hsp=None, hpi=None, elp=None, point_names=None,
         hsp = np.array(hsp) if len(hsp) else None
         elp = np.array(elp) if len(elp) else None
         coord_frame = 'head'
+    elif egi is not None:
+        if not all(x is None for x in (hsp, hpi, elp, point_names, fif)):
+            raise ValueError('hsp, hpi, elp, point_names, fif must all be '
+                             'None if egi is not None')
+        _check_fname(egi, overwrite=True, must_exist=True)
+
+        root = ElementTree.parse(egi).getroot()
+        ns = root.tag[root.tag.index('{'):root.tag.index('}') + 1]
+        sensors = root.find('%ssensorLayout/%ssensors' % (ns, ns))
+        fids = dict()
+        dig_ch_pos = dict()
+
+        fid_name_map = {'Nasion': 'nasion',
+                        'Right periauricular point': 'rpa',
+                        'Left periauricular point': 'lpa'}
+
+        for s in sensors:
+            name, number, kind = s[0].text, int(s[1].text), int(s[2].text)
+            coordinates = np.array([float(s[3].text), float(s[4].text),
+                                    float(s[5].text)])
+            # EEG Channels
+            if kind == 0:
+                dig_ch_pos['EEG %03d' % number] = coordinates
+            # Reference
+            elif kind == 1:
+                dig_ch_pos['EEG %03d' %
+                           (len(dig_ch_pos.keys()) + 1)] = coordinates
+            # Fiducials
+            elif kind == 2:
+                fid_name = fid_name_map[name]
+                fids[fid_name] = coordinates
+            # Unknown
+            else:
+                warn('Unknown sensor type %s detected. Skipping sensor...'
+                     'Proceed with caution!' % kind)
+
+        fids = [fids[key] for key in ('nasion', 'lpa', 'rpa')]
```

Comment:
We can add a check for missing ones later if this fails in an uninformative way. In the meantime we'll assume YAGNI. So this is good as is